providers/aws/fargate/pod.go

```diff
@@ -104,6 +108,11 @@ func NewPod(cluster *Cluster, pod *corev1.Pod) (*Pod, error) {
 	taskDef.Cpu = aws.String(strconv.Itoa(int(fgPod.taskCPU)))
 	taskDef.Memory = aws.String(strconv.Itoa(int(fgPod.taskMemory)))
 
+	if val, ok := pod.Annotations[taskRoleAnnotation]; ok {
+		taskDef.TaskRoleArn = aws.String(val)
+	}
+	fgPod.taskRoleArn = taskDef.TaskRoleArn
```

Comment:
Can we please move this inside the if block? fdPod.taskRoleArn is only set if there is an annotation (ok == true), so shouldn't it be also inside the if block? Simply fdPod.taskRoleArn = val